src/gui/torrentmodel.cpp

```diff
@@ -283,6 +293,131 @@ Qt::ItemFlags TorrentModel::flags(const QModelIndex &index) const
     return QAbstractListModel::flags(index) | Qt::ItemIsEditable;
 }
 
+QVariant TorrentModel::getTooltip(const QModelIndex& idx, const BitTorrent::TorrentHandle *torrent) const
+{
+    TransferListWidget *pWdiget = dynamic_cast<TransferListWidget *>(m_parent);
+
+    if (!pWdiget || !m_parent) return QVariant(""); // with QVariant() any currently visible tooltip does not go away
+
+    bool isHideState = true;
+    if (Preferences::instance()->getHideZeroComboValues() == 1) {  // paused torrents only
+        QModelIndex statusIndex = idx.sibling(idx.row(), TorrentModel::TR_STATUS);
+        if (statusIndex.data().toInt() != BitTorrent::TorrentState::PausedDownloading)
+            isHideState = false;
+    }
+    const bool hideValues = Preferences::instance()->getHideZeroValues() & isHideState;
+    QFontMetrics fontmetrics(pWdiget->font());
+    int textw = 0;
+    QVariant text;
+
+    switch (idx.column()) {
+    case TorrentModel::TR_NAME:
+        textw += pWdiget->style()->pixelMetric(QStyle::PM_ListViewIconSize) + fontmetrics.width(" "); // icon width + margin
```

Comment:
add `// fallthrough` if you intend to do so (in order to suppress warning in Coverity)